modules/sdk-lib-mpc/src/tss/ecdsa-dkls/types.ts

```diff
@@ -0,0 +1,96 @@
+// Broadcast message meant to be sent to multiple parties
+interface BroadcastMessage<T> {
+  payload: T;
+  from: number;
+}
+
+// P2P message meant to be sent to a specific party
+interface P2PMessage<T, G> {
+  payload: T;
+  from: number;
+  commitment?: G;
+  to: number;
+}
+
+export enum DkgState {
+  Uninitialized = 0,
+  Round1,
+  Round2,
+  Round3,
+  Round4,
+  Complete,
+  InvalidState,
+}
+
+export type AuthEncMessage = {
+  encryptedMessage: string;
+  signature: string;
+};
+export type PartyGpgKey = {
+  partyId: number;
+  gpgKey: string;
+};
+export type SerializedBroadcastMessage = BroadcastMessage<string>;
+export type DeserializedBroadcastMessage = BroadcastMessage<Uint8Array>;
+export type SerializedP2PMessage = P2PMessage<string, string>;
+export type DeserializedP2PMessage = P2PMessage<Uint8Array, Uint8Array>;
+export type AuthEncP2PMessage = P2PMessage<AuthEncMessage, string>;
+export type AuthBroadcastMessage = BroadcastMessage<AuthEncMessage>;
```

Comment:
`AuthBroadcastMessage` being parameterized to `AuthEncMessage` is a little confusing, maybe apart from `AuthEncMessage`, we need another type `AuthMessage`.

Reply:
done